### DISCUSSION TITLE
Docs Addon: Possible to get the current args in transformSource?

### DISCUSSION THREAD

**Discussion by nolledgeable** [storybookjs/storybook#17268] [Help]
Hi guys,

Is there anyway to get the currently set args (via the controls tab) inside the `transformSource` function in the docs parameters? My goal is to run the `originalStoryFn` with these args and render that to an HTML string using `renderToString` from `react-dom/server`. I will format and show that HTML string below the original React snippet.

The simplified implementation looks like this:

```js
export default {
  title: "MyStory",
  parameters: {
    docs: {
      transformSource( snippet, storyContext ) {
        // Can I get current args as set in controls tab here?
        const currentArgs = storyContext.somethingSomething;
        // To use with the originalStoryFn.
        const story = storyContext.originalStoryFn( currentArgs );
        // Render to HTML string.
        const html = ReactDOMServer.renderToString( story );
        // Return both original React snippet and additional HTML string as source.
        return `
          // React snippet
          ${ snippet }

          // HTML snippet
          ${ html }
        `;
      },
    },
  },
};
```

In `storyContext` I do have the `initialArgs` available, but not the current args as set in the controls tab. However, they must be available in this flow since the original React snippet passed to `transformSource` uses the current args. Does anyone have an idea how to get these args in this context? Should I select from some store based on story `id` for instance?

**Comment by oemer-aran** [comment]
Seems like it simply worked with `storyContext.args` previously: https://github.com/storybookjs/storybook/issues/13079#issuecomment-725396542

Is this no longer supported?

**Comment by oemer-aran** [comment]
Any update/clarification?

**Comment by tannguyenad** [comment]
Im having the same use case, would to update the code snippet according to any change on controls table. Is there a workaround? Tried to use `storyContext.parameters.args` but it just returned the initial args.

Here is the relevant issue: https://github.com/storybookjs/storybook/issues/17341

**Comment by FuriouZz** [comment]
I use a trick by passing the arguments via `ctx.originalStoryFn` as a temporary solution.

```ts
const Template = (args, ctx) => {
  ctx.originalStoryFn.currentArgs = args;
  return <Component {...args} />
};
```

**Comment by francisco-lgpc** [comment]
Any updates?

Thanks for the workaround @FuriouZz 

What I ended up doing was adding a decorator to my preview.js

```javascript
const withCurrentArgs = (fn, context) => {
  const story = fn(context);

  context.parameters.currentArgs = story.props;

  return story;
};
```

**Reply by trickpattyFH20** [reply]
can you show the full preview.js code for this? @francisco-lgpc
